pkg/chains/storage/grafeas/grafeas_test.go

```diff
@@ -0,0 +1,350 @@
+	pb.UnimplementedGrafeasV1Beta1Server
+
+	// Assume there is only one project for storing notes and occurences
+	occurences map[string]*pb.Occurrence
+	notes      map[string]*pb.Note
+}
+
+var grafeasServer mockGrafeasV1Beta1Server
+
+func TestBackend_StorePayload(t *testing.T) {
+	tests := []testConfig{
+		// test 1
+		{
+			name: "no error, simplesining and intoto",
+			args: args{
+				tr: &v1beta1.TaskRun{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "foo1",
+						Name:      "bar1",
+						UID:       types.UID("uid1"),
+					},
+					Status: v1beta1.TaskRunStatus{
+						TaskRunStatusFields: v1beta1.TaskRunStatusFields{
+							TaskRunResults: []v1beta1.TaskRunResult{
+								{Name: "IMAGE_DIGEST", Value: "sha256:cfe4f0bf41c80609214f9b8ec0408b1afb28b3ced343b944aaa05d47caba3e00"},
+								{Name: "IMAGE_URL", Value: "gcr.io/test/kaniko-chains1"},
+							},
+						},
+					},
+				},
+				trPayload:    []byte("taskrun payload"),
+				trSignature:  "taskrun signature",
+				trOpts:       config.StorageOpts{Key: "taskrun.uuid", PayloadFormat: formats.PayloadTypeInTotoIte6},
+				ociPayload:   []byte("oci payload"),
+				ociSignature: "oci signature",
+				ociOpts:      config.StorageOpts{Key: "cfe4f0bf41c8", PayloadFormat: formats.PayloadTypeSimpleSigning},
+			},
+			wantErr: false,
+		},
+	}
+
+	conn, client, err := setupConnection()
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+
+			if err != nil {
+				t.Error(err)
+			}
+
+			backend := Backend{
+				logger: logtesting.TestLogger(t),
+				tr:     test.args.tr,
+				client: client,
+				cfg: config.Config{
+					Storage: config.StorageConfigs{
+						Grafeas: config.GrafeasConfig{
+							ProjectID: "test-project",
+							NoteID:    "test-note",
+						},
+					},
+				},
+			}
+
+			ctx := context.Background()
+
+			// test if the attestation of the **taskrun** artifact can be successfully stored into grafeas server
+			// and test if payloads and signatures inside the attestation can be retrieved.
+			testInterface(ctx, t, test, backend, test.args.trPayload, test.args.trSignature, test.args.trOpts)
+
+			// test if the attestation of the **oci** artifact can be successfully stored into grafeas server
+			// and test if payloads and signatures inside the attestation can be retrieved.
+			testInterface(ctx, t, test, backend, test.args.ociPayload, test.args.ociSignature, test.args.ociOpts)
+
+			// test if all occurrences generated from `StorePayload` are what we expect.
+			testListOccurrences(ctx, t, backend)
+		})
+	}
+
+	// close connection
+	conn.Close()
+}
+
+// test attestation storage and retrieval
+func testInterface(ctx context.Context, t *testing.T, test testConfig, backend Backend, payload []byte, signature string, opts config.StorageOpts) {
+	if err := backend.StorePayload(ctx, payload, signature, opts); (err != nil) != test.wantErr {
+		t.Errorf("Backend.StorePayload() error = %v, wantErr %v", err, test.wantErr)
```

Comment:
```suggestion
		t.Fatalf("Backend.StorePayload() error = %v, wantErr %v", err, test.wantErr)
```

Fatal will fail the test and stop execution.
Error will fail the test and continue execution.